src/gui/DatabaseOpenWidget.cpp

```diff
@@ -251,6 +251,7 @@ void DatabaseOpenWidget::openDatabase()
     const auto databaseKey = buildDatabaseKey();
     if (!databaseKey) {
         setUserInteractionLock(false);
+        resetQuickUnlock();
```

Comment:
I believe this will even reset quick unlock when you merely canceled the interaction. To me that is not a desired behavior. We should capture when an explicit error occurs with quick unlock then reset on that case alone.

Reply:
Yes, that's right. As a shortcut for testing I've been clicking cancel in the Windows Hello dialog, and indeed it will reset in that case. I agree this isn't the best and I'll try to detect errors here, or try to detect cancellation, or something.

Reply:
I pushed a commit adding a bunch of plumbing for detecting an explicit cancellation, and not resetting in that case. However, my changes might have done more harm than good, especially since I'm only piling onto tech debt. I'm totally fine with just backing out all commits in this PR except for the first.